### PR TITLE
Simulate user registration (GIM-379)

### PR DESCRIPTION
**Pull request by Cito**
This PR makes the registration a part of the tested user journeys, instead of adding users directly to the database, which is not possible in black-box testing mode.

A new config setting `use_api_gateway` for black-box testing has been added, which will be fully implemented in GIM-323.

### CODE REVIEW

File: steps/test_26_dataset_summary.py [hunk 1]
```diff
@@ -44,7 +44,7 @@
         },
         "files_summary": {
             "count": 20,
-            "stats": {"format": [{"value": "FASTQ", "count": 20}], "size": 114708},
+            "stats": {"format": [{"value": "FASTQ", "count": 20}], "size": 58720716},
```
Comment (lkuchenb) [inline steps/test_26_dataset_summary.py:47]: Is this change related to the rest of the patch / intentional?

**Review by lkuchenb — APPROVED** [review]
Thanks, LGTM, just verify the inline comment before merging.

File: steps/test_26_dataset_summary.py (hunk 1 above)
Comment (Cito) [reply steps/test_26_dataset_summary.py:47]: No, it's not related. It fixes the summary that was broken after PR #48 when metldata was updated.
